src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -96,6 +98,19 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     }
   }
 
+  @Override
+  public ResponseEntity<SearchMetadataModel> upsertSearchMetadata(
+      @PathVariable("id") String id,
+      @Valid @RequestBody SearchMetadataRequest searchMetadataRequest) {
+    try {
+      SearchMetadataModel searchMetadataModel = new SearchMetadataModel();
```

Comment:
This will need an authz check. I think we should use `IamAction.UPDATE_SNAPSHOT`. The description for this action is `Update snapshot information` which seems appropriate, and `steward` and `custodian` can perform this action. It doesn't appear to be currently used in TDR.